src/cameraGeom/TransformMap.cc

```diff
@@ -84,12 +88,160 @@ std::shared_ptr<ast::Mapping const> TransformMap::_getMapping(CameraSys const &f
 
 size_t TransformMap::size() const noexcept { return _frameIds.size(); }
 
-TransformMap::TransformMap(std::unique_ptr<ast::FrameSet> && transforms, CameraSysFrameIdMap && frameIds) :
+TransformMap::TransformMap(std::unique_ptr<ast::FrameSet> && transforms,
+                           CameraSysFrameIdMap && frameIds,
+                           std::vector<std::pair<int, int>> && canonicalConnections) :
     _transforms(std::move(transforms)),
-    _frameIds(std::move(frameIds))
+    _frameIds(std::move(frameIds)),
+    _canonicalConnections(std::move(canonicalConnections))
 {}
 
 
+namespace {
+
+struct PersistenceHelper {
+
+    static PersistenceHelper const & get() {
+        static PersistenceHelper const instance;
+        return instance;
+    }
+
+    table::Schema sysSchema;
+    table::Key<std::string> sysName;
+    table::Key<std::string> detectorName;
+    table::Key<int> id;
+
+    table::Schema connectionSchema;
+    table::Key<int> from;
+    table::Key<int> to;
+    table::Key<int> transform;
+
+    CameraSys makeCameraSys(table::BaseRecord const & record) const {
+        return CameraSys(record.get(sysName), record.get(detectorName));
+    }
+
+private:
+
+    PersistenceHelper() :
+        sysSchema(),
+        sysName(sysSchema.addField<std::string>("sysName", "Camera coordinate system name", "", 0)),
+        detectorName(sysSchema.addField<std::string>("detectorName",
+                                                     "Camera coordinate system detector name", "", 0)),
+        id(sysSchema.addField<int>("id", "AST ID of the Frame for the CameraSys", "")),
+        connectionSchema(),
+        from(connectionSchema.addField<int>("from", "AST ID of the Frame this transform maps from.", "")),
+        to(connectionSchema.addField<int>("to", "AST ID of the Frame this transform maps to.", "")),
+        transform(connectionSchema.addField<int>("transform", "Archive ID of the transform.", ""))
+    {
+        sysSchema.getCitizen().markPersistent();
+        connectionSchema.getCitizen().markPersistent();
+    }
+
+    PersistenceHelper(PersistenceHelper const &) = delete;
+    PersistenceHelper(PersistenceHelper &&) = delete;
+
+    PersistenceHelper & operator=(PersistenceHelper const &) = delete;
+    PersistenceHelper & operator=(PersistenceHelper &&) = delete;
+
+};
+
+std::string makeFrameName(CameraSys const & sys) {
+    std::string r = "Ident=" + sys.getSysName();
+    if (sys.hasDetectorName()) {
+        r += "_";
+        r += sys.getDetectorName();
+    }
+    return r;
+}
+
+} // anonymous
```

Comment:
space space // namespace?

Reply:
Done.

As a side note, I am a bit worried about the degree to which usage of clang-format's particular style seems to be becoming an implied MUST rather than the [MAY](https://developer.lsst.io/cpp/style.html#layout-may-be-automated-with-clang-format-using-the-lsst-configuration) that was actually RFC'd.  But this is not the hill I want my clang-format-free-workflow to die on, and I certainly can't pretend I have a good reason to differ from it on this particular case.